unlock-app/src/hooks/useNetwork.ts

```diff
@@ -0,0 +1,32 @@
+import { useMutation } from 'react-query'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useConfig } from '~/utils/withConfig'
+
+export const useNetwork = () => {
+  const { networks } = useConfig()
+  const { network: connectedNetwork, changeNetwork } = useAuth()
+
+  const onChangeNetwork = async ({ network }: any) => {
+    return await changeNetwork(networks[network])
+  }
+
+  const changeNetworkMutation = useMutation(onChangeNetwork)
+  const switchToNetworkBeforeAction = async (
+    network: number,
+    callback: () => void
+  ) => {
+    const differentNetwork = network != connectedNetwork
+    if (differentNetwork) {
+      await changeNetworkMutation.mutate({
+        network,
+      })
+      await callback()
+    } else {
+      await callback()
+    }
+  }
+
+  return {
+    switchToNetworkBeforeAction,
+  }
```

Comment:
As discussed, let's not add hook for this and re-use existing hook since it's a minor change in function.